cmd/containerboot/serve.go

```diff
@@ -72,8 +72,10 @@ func watchServeConfigChanges(ctx context.Context, path string, cdChanged <-chan
 		if err := updateServeConfig(ctx, sc, certDomain, lc); err != nil {
 			log.Fatalf("serve proxy: error updating serve config: %v", err)
 		}
-		if err := kc.storeHTTPSEndpoint(ctx, certDomain); err != nil {
-			log.Fatalf("serve proxy: error storing HTTPS endpoint: %v", err)
+		if kc != nil {
```

Comment:
If I'm reading the code paths correctly, it looks like we'll also need this nil-guard in `watchServeConfigChanges`

Reply:
Technically it also seems possible to enable `cfg.EgressSvcsCfgPath` outside of k8s and hit a similar panic, but we should probably add an explicit validation error in `settings.go` for that

Reply:
Oh sorry I'm an idiot, disregard my first comment

Reply:
What do you mean? This _is_ in `watchServeConfigChanges`

Reply:
fwiw, I tested this e2e with https://github.com/tailscale-dev/docker-guide-code-examples/tree/main/04-ts-mealie

Reply:
> an explicit validation error in settings.go for that

yeah definitely could add more validation checks 👍🏼

Reply:
I'll make myself a coffee and return with my brain attached 😅

Reply:
(I'll take a look at extra validations separately,but will only cherry-pick this)

Reply:
![Alt Text](https://media.tenor.com/GvYdH_A6DH0AAAAM/coffee-pour.gif)